org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/MavenCompletionParticipant.java

```diff
@@ -83,26 +88,28 @@ public void onTagOpen(ICompletionRequest request, ICompletionResponse response)
 	@Override
 	public void onXMLContent(ICompletionRequest request, ICompletionResponse response) throws Exception {
 		DOMElement parent = request.getParentElement();
-
 		if (parent == null || parent.getLocalName() == null) {
 			return;
 		}
-		// TODO: These two switch cases should be combined into one
-		if (parent.getParentElement() != null) {
-			switch (parent.getParentElement().getLocalName()) {
-			case "parent":
-				collectParentCompletion(request, response);
-				break;
-			case "plugin":
-				collectRemotePluginGAVCompletion(request, response);
-				break;
-			case "dependency":
-				collectRemoteGAVCompletion(request, response);
-				break;
-			default:
-				break;
-			}
-		}
+		DOMElement grandParent = parent.getParentElement();
+		boolean isPlugin = "plugin".equals(parent.getLocalName()) || (grandParent != null && "plugin".equals(grandParent.getLocalName()));
+		boolean isParentDeclaration = "parent".equals(parent.getLocalName()) || (grandParent != null && "parent".equals(grandParent.getLocalName()));
+		Optional<String> groupId = grandParent == null ? Optional.empty() : grandParent.getChildren().stream()
+				.filter(DOMNode::isElement)
+				.filter(node -> "groupId".equals(node.getLocalName()))
+				.flatMap(node -> node.getChildren().stream())
+				.map(DOMNode::getTextContent)
+				.filter(Objects::nonNull)
+				.filter(Predicate.not(String::isBlank))
```

Comment:
`String.isBlank()` is a Java 11 feature. Maybe this could be replaced by something like `.map(String::trim).filter(Predicate.not(String::isEmpty))`?